docs/pages/database-access/guides/clickhouse-self-hosted.mdx

```diff
@@ -0,0 +1,213 @@
+       <caConfig>/path/to/server.cas</caConfig>
+       <certificateFile>/path/to/server.crt</certificateFile>
+       <verificationMode>strict</verificationMode>
+    </server>
+</openSSL>
+```
+
+See [SSL-TLS](https://clickhouse.com/docs/en/guides/sre/configuring-ssl) ClickHouse documentation for more details.
+
+Additionally, your ClickHouse database user accounts must be configured to require a valid client certificate:
+
+```sql
+CREATE USER alice IDENTIFIED WITH ssl_certificate CN 'alice';
+```
+
+By default, the created user may not have access to anything and won't be able to connect, so let's grant it some permissions:
+
+```sql
+GRANT ALL ON *.* TO alice;
+```
+
+## Step 4/5. Configure and start the Database Service
+Install and configure Teleport on the host or Kubernetes cluster where you will run the Teleport Database Service:
+
+<Tabs>
+<TabItem label="Linux Server">
+
+(!docs/pages/includes/install-linux.mdx!)
+
+Configure and start the Teleport service:
+
+
+(!docs/pages/includes/database-access/db-configure-start-clickhouse.mdx dbProtocol="clickhouse-http" dbName="example-clickhouse-http" databaseAddress="clickhouse.example.com:8443" !)
+
+</TabItem>
+<TabItem label="Kubernetes Cluster">
+  Teleport provides Helm charts for installing the Teleport Database Service in Kubernetes Clusters.
+
+  (!docs/pages/kubernetes-access/helm/includes/helm-repo-add.mdx!)
+
+  (!docs/pages/includes/database-access/db-helm-install.mdx dbName="example-clickhouse-http" dbProtocol="clickhouse-http" databaseAddress="clickhouse.example.com:8443" dbName="clickhouse-http" !)
+
+</TabItem>
+</Tabs>
+
+(!docs/pages/includes/database-access/multiple-instances-tip.mdx !)
+
+## Step 5/5. Connect
+
+Once the Database Service has joined the cluster, log in to see the available
+databases:
+
+
+<Tabs>
+  <TabItem label="ClickHouse HTTP Protocol">
+
+```code
+$ tsh login --proxy=<Var name="proxy-address" /> --user=testuser
+$ tsh db ls
+# Name                                        Description Allowed Users Labels  Connect
+# ------------------------------------------- ----------- ------------- ------- ------------------------------
+# example- clickhouse-http (user: alice)             [*]           env=dev tsh db connect clickhouse-http
+```
+
+Create an authenticated proxy tunnel for allowing to connect to the ClickHouse Database GUI database client send a request via `curl`:
+
+```code
+$ tsh proxy db --tunnel example-clickhouse-http
+# Started authenticated tunnel for the Clickhouse (HTTP) database "clickhouse-http" in cluster "ice-berg.dev" on 127.0.0.1:59215.
+# To avoid port randomization, you can choose the listening port using the --port flag.
+#
+# Use the following command to connect to the database or to the address above using other database GUI/CLI clients:
+#   $ curl http://localhost:59215/
+```
+
+To test the connection you can run the following command:
+
+```code
+$ echo 'select 1;' | curl http://localhost:59215/  --data-binary @-
+# 1
```

Comment:
nit: i assume we use `select 1` to test if connection is established but maybe we can give a better/more realistic example statement here